packages/community/src/llm/google/index.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  GenerateContentResponse,
+  GoogleGenAI,
+  Modality,
+  type Blob,
+  type GenerateContentConfig,
+  type GoogleGenAIOptions,
+} from "@google/genai";
+import {
+  ToolCallLLM,
+  type ChatResponse,
+  type ChatResponseChunk,
+  type CompletionResponse,
+  type LLMChatParamsNonStreaming,
+  type LLMChatParamsStreaming,
+  type LLMCompletionParamsNonStreaming,
+  type LLMCompletionParamsStreaming,
+  type LLMMetadata,
+  type ToolCall,
+  type ToolCallLLMMessageOptions,
+} from "@llamaindex/core/llms";
+export { Modality };
+
+import { streamConverter } from "@llamaindex/core/utils";
+
+import { wrapLLMEvent } from "@llamaindex/core/decorator";
+import type { JSONObject } from "@llamaindex/core/global";
+import { GEMINI_MODEL } from "@llamaindex/google";
+import { mapMessageContentToMessageContentDetails } from "../utils";
+import {
+  mapChatMessagesToGoogleFunctions,
+  mapChatMessagesToGoogleMessages,
+  mapMessageContentDetailToGooglePart,
+} from "./utils";
+
+export type GoogleAdditionalChatOptions = { config: GenerateContentConfig };
+
+export type GoogleChatStreamResponse = AsyncIterable<
+  ChatResponseChunk<
+    ToolCallLLMMessageOptions & {
+      data?: Blob[];
+    }
+  >
+>;
+
+export type GoogleChatParamsStreaming = LLMChatParamsStreaming<
+  GoogleAdditionalChatOptions,
+  ToolCallLLMMessageOptions
+>;
+
+export type GoogleChatParamsNonStreaming = LLMChatParamsNonStreaming<
+  GoogleAdditionalChatOptions,
+  ToolCallLLMMessageOptions
+>;
+
+export type GoogleChatNonStreamResponse =
+  ChatResponse<ToolCallLLMMessageOptions>;
+
+export const getGoogleStudioInlineData = (
+  response: GenerateContentResponse,
+): Blob[] => {
+  return response.candidates
+    ?.flatMap((candidate) => candidate.content?.parts)
+    .map((part) => part?.inlineData)
+    .filter((data) => data) as Blob[];
+};
+
+const SUPPORT_TOOL_CALL_MODELS: GEMINI_MODEL[] = [
+  GEMINI_MODEL.GEMINI_PRO,
+  GEMINI_MODEL.GEMINI_PRO_VISION,
+  GEMINI_MODEL.GEMINI_PRO_1_5_PRO_PREVIEW,
+  GEMINI_MODEL.GEMINI_PRO_1_5_FLASH_PREVIEW,
+  GEMINI_MODEL.GEMINI_PRO_1_5,
+  GEMINI_MODEL.GEMINI_PRO_1_5_FLASH,
+  GEMINI_MODEL.GEMINI_PRO_LATEST,
+  GEMINI_MODEL.GEMINI_PRO_FLASH_LATEST,
+  GEMINI_MODEL.GEMINI_PRO_1_5_LATEST,
+  GEMINI_MODEL.GEMINI_PRO_1_5_FLASH_LATEST,
+  GEMINI_MODEL.GEMINI_2_0_FLASH_EXPERIMENTAL,
+  GEMINI_MODEL.GEMINI_2_0_FLASH,
+  GEMINI_MODEL.GEMINI_2_0_PRO_EXPERIMENTAL,
+];
+
+const DEFAULT_GEMINI_PARAMS = {
+  model: GEMINI_MODEL.GEMINI_PRO,
+  temperature: 0.1,
+  topP: 1,
+  maxTokens: undefined,
+};
+
+export type GoogleModelParams = {
+  model: GEMINI_MODEL;
+  temperature?: number;
+  topP?: number;
+  maxTokens?: number;
+};
+
+export type GoogleParams = GoogleGenAIOptions & GoogleModelParams;
+
+export class GoogleStudio extends ToolCallLLM<GoogleAdditionalChatOptions> {
```

Comment:
@parhammmm thanks for your PR! Looks great, but I am not sure putting it in the community package is the best way.

The community package doesn't make a lot of sense to me anymore now that we have external dependencies wrapped in our own provider packages.
It contains only AWS bedrock code, so it better should be in a new @llamaindex/aws package.

About the new `GoogleStudio` class - do you think we can just move it to @llamaindex/google without impacting the existing google classes?

Reply:
@marcusschiesser sure I'll update this PR and move the code inside the google provider

Reply:
@marcusschiesser there's a version clash between `js-genai` and the mix of vertex sdk and generative-ai-js

I think it's best just to have it in the community package for the time being until a full switch

Reply:
@marcusschiesser turns out it's the model itself that's not behaving as expected and not the libraries

`GoogleStudio` is now in `@llamaindex/google` and can be reviewed